src/pages/blog/optimizing-emotion.mdx

```diff
@@ -0,0 +1,293 @@
+
+## Just how much work is Emotion doing?
+
+Just like at the build time of a CSS preprocessor, Emotion has to parse the style passed into it. Emotion _does not_ simply pass its input strings into a `<style>` tag, it must parse the style strings into an object tree. The reason your style strings must be parsed by Emotion before the browser is because Emotion offers SCSS-like nesting syntax as it uses [`stylis`](https://www.npmjs.com/package/stylis) as it's underlying stylsheet language.
+
+Once the parsing is complete, it then generates a minified CSS output and inserts that value into a `<style>` tag in the `<head>` of the document. If you are using SSR then the content will be in the tag as a normal [Text](https://developer.mozilla.org/en-US/docs/Web/API/Text) node (which is how it content would load if you hand-wrote CSS into a `<style>` element). For SPA styles, they will be added rule-by-rule directly to a live [`CSSRuleList`](https://developer.mozilla.org/en-US/docs/Web/API/CSSRuleList) of an empty `<style>` tag for improved performance in cases of more frequent modifications of style rules.
+
+The generic `@emotion/css` library then returns a string of a random-hashed className that can be used to reference the styles that were just generated. The `@emotion/react` library handles the order-of-operations here slightly differently by only running style rule insertions when components are being mounted and memoizing the construction of these styles for future renders. There are performance advantages to using `@emotion/react` over `@emotion/css` in React applications but in the end, it always has to do all of these steps.
+
+_This is a lot of work to do at runtime._ The fact that most applications using Emotion don't show obvious perfomance buckling or dropped frames is impressive to the quality of the engineering at work in Emotion. However, I think it's important to understand here that using Emotion is never necessary and is always a tradeoff for gains in the developer experience at the risk of lost performance. There is nothing that you can do with Emotion that you can't do without it but it enhances code readability and increases developer velocity with the code colocation that it allows for.
+
+## How can we optimize our use of Emotion in React applications?
+
+### Opimization I. Avoid doing construction of styles in render functions.
+
+Now that it's clear how much work Emotion has to do at runtime, it becomes far more obvious that we want to do this work as infrequently as possible. So this means avoiding using Emotion style constructions in renders as much as we possibly can. Despite almost every example in [the Emotion documentation](https://emotion.sh/docs/introduction) showing style constructions in render functions, this is not the best practice. The documentation, to me, reads as a bit too focussed on what is possible with Emotion and not enough on what would be best practice for performance.
```

Comment:
I thought `focussed` needed to be one `s` but apparently both are correct!

Reply:
You know I thought I did spell it wrong at a point and I just Google'd "focussed" and when it didn't say "Did you mean 'focused'?" I assumed I was right. But evidently, this is the British/Australian spelling so I'm going to swap it to one S for consistency.